### PR TITLE
Empty Request and Response Structs

### PR DESCRIPTION
**Pull request by ajberchek**
Since there are many endpoints which require "no parameters" for the request and send back "no body" in the response, I thought it would be fitting to create an Empty request and Empty response struct with corresponding to_json and from_json functions to easily integrate with existing code in mtxclient. 

Individual endpoints (e.g. logout) can simply be integrated by adding a `using Logout = Empty;` in the files which define Empty, then can be used like any other matrix-struct.

### CODE REVIEW

**Comment by mujx** [comment]
Sorry about the lint. I'll move it to a linux build because the one in macOS seems to be failing.

**Comment by ajberchek** [comment]
It looks like it works now, it was conflicting with itself. Running `make lint` caused the empty structs to still be formatted incorrectly compared to the Travis Lint, so I updated it to the Travis suggested version, and it works :+1:

**Comment by mujx** [comment]
Yeah, they should have been the same. Are you using clang-format v5?

**Comment by ajberchek** [comment]
No, I guess I am much more outdated than I thought lol. I am at the most up to date version in the ubuntu repos - v4 I'll try to install a newer version in line with Travis' before running `make lint` and pushing again.

File: src/responses/empty.cpp [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+namespace mtx {
+namespace responses {
+
+//Provides a deserialization function to use when empty responses are returned from the server
+void
+from_json(const json &obj, Empty &response)
+{
+	//Cast these to void so the compiler doesn't complain about unused variables
+	(void)obj;
```
Comment (mujx) [inline src/responses/empty.cpp]: You can leave the body empty (there isn't much to do anyway) and omit the parameters from the function to stop the errors from the compiler.

Comment (ajberchek) [reply src/responses/empty.cpp]: It won't cause any incompatibility issues with other calls to from_json on Empty structs in mtxclient?

Comment (mujx) [reply src/responses/empty.cpp]: No, it would still be valid. Of course you can run some tests and see for yourself.

Comment (ajberchek) [reply src/responses/empty.cpp]: Oh, I misunderstood, I thought you meant I should remove the parameters and their types, and I had run into an error regarding that issue before.

I removed the parameter declarations, left their types, and everything works just like you said. Thanks for the suggestion!

**Comment by ajberchek** [comment]
I now have clang-format v5, so lint should no longer cause any discrepancies.

**Comment by mujx** [comment]
Thanks again!

**Pull request merged**